File: 1-arrays-and-strings/reverseString2.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        
        Time: O(N)
        Space: O(1)
        """
        left, right = 0, len(s) - 1
        
        while left < right:
            s[left], s[right] = s[right], s[left]
            left, right = left + 1, right -1
        
        return s
    

File: 1-arrays-and-strings/reverseString2.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 * 
 * Time: O(N)
 * Space: O(1)
 */
var reverseString = function(s) {
  for (let i = 0, j = s.length -1; i < j; i++, j--) {
      [s[i], s[j]] = [s[j], s[i]];
  }
  
  return s;
};
